counsel.el

```diff
@@ -201,6 +203,23 @@ respectively."
     (set-process-filter proc (or filter #'counsel--async-filter))
     proc))
 
+(defcustom counsel-async-command-delay 0.1
+  "The amount of seconds to wait until spawn next async command."
+  :type 'float
+  :group 'counsel)
+
+(defun counsel--async-command-delay (&rest args)
+  "Like `counsel--async-command' with same ARGS but run after a little delay.
+This's to help prevent spamming subcommand."
```

Comment:
Nit: it would be nice to mention that this function "debounces" the calls to `counsel--async-timer`.

```el
  "Like `counsel--async-command', with same ARGS, but debounced.
Calls to `counsel--async-command' are separated by at least
`counsel-async-command-delay' seconds, so as to avoid issues
caused by spawning too many subprocesses too quickly."
```